ballerina-tests/tests/http_cache_config_annotation_test.bal

```diff
@@ -0,0 +1,303 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/test;
+import ballerina/lang.runtime as runtime;
+import ballerina/http;
+import ballerina/crypto;
+
+http:Client cacheClientEP = check new("http://localhost:" + cacheAnnotationTestPort1.toString(), { cache: { enabled: false }});
+
+http:Client cacheBackendEP = check new("http://localhost:" + cacheAnnotationTestPort2.toString(), { cache: { isShared: true } });
+
+int numberOfProxyHitsNew = 0;
+int noCacheHitCountNew = 0;
+int maxAgeHitCountNew = 0;
+int numberOfHitsNew = 0;
+int statusHits = 0;
+xml maxAgePayload1 = xml `<message>before cache expiration</message>`;
+xml maxAgePayload2 = xml `<message>after cache expiration</message>`;
+string errorBody = "Error";
+string mustRevalidatePayload1 = "Hello, World!";
+byte[] mustRevalidatePayload2 = "Hello, New World!".toBytes();
+json nocachePayload1 = { "message": "1st response" };
+json nocachePayload2 = { "message": "2nd response" };
+http:Ok ok = {body : mustRevalidatePayload1};
+http:InternalServerError err = {body : errorBody};
+
+service / on new http:Listener(cacheAnnotationTestPort1) {
+
+    resource function get noCache(http:Request req) returns http:Response|http:InternalServerError {
+        http:Response|error response = cacheBackendEP->forward("/nocacheBE", req);
+        if (response is http:Response) {
```

Comment:
parentheses are not required around the condition so we can remove parentheses around if condition. 

 https://docs.google.com/document/d/1y6QVqaZzZt9jMpYV4jP5WRS_W_KoC4y40Uuoh1ALu8E/edit#